internal/model/rwfolder.go

```diff
@@ -1222,8 +1223,30 @@ func (p *rwFolder) performFinish(state *sharedPullerState) error {
 		}
 	}
 
-	// Set the correct timestamp on the new file
+	// Verify if a file will be overwritten and if we are up to date about its current mtime
 	t := time.Unix(state.file.Modified, 0)
+	existingInfo, existingErr := os.Lstat(state.realName)
+	if existingErr != nil {
```

Comment:
Also, why do we need this? We havr the mtime (or curFile if we want) in the sharedstate?